android/src/main/java/com/doublesymmetry/trackplayer/service/MusicService.kt

```diff
@@ -400,6 +401,7 @@ class MusicService : HeadlessJsTaskService() {
                         }
 
                         stopSelf()
+                        exitProcess(0)
```

Comment:
`onTaskRemoved` is only called once, which means that the notification would not disappear next time the app is removed from recents. This is probably related to the process remaining running.

`exitProcess` ensures the process stops. Not the most elegant solution, but it works.

Reply:
I discovered a serious flaw with this approach. Calling `TrackPlayer.reset()` causes this to be triggered and kills the app, which means there is no longer any way to stop the player during regular use.

Reply:
@mmmoussa https://github.com/doublesymmetry/react-native-track-player/issues/1768

Reply:
@mpivchev I noticed you are calling both startService() and bindService(). 
https://github.com/doublesymmetry/react-native-track-player/blob/e742959cc1d697d69daeade39830155bce2ccde7/android/src/main/java/com/doublesymmetry/trackplayer/module/MusicModule.kt#L181-L184

Maybe this explanation on SO helps on how to stop the service? https://stackoverflow.com/a/17178363/3229423